cmd/kubemark/hollow-node.go

```diff
@@ -145,11 +159,75 @@ func run(config *hollowNodeConfig) {
 	}
 
 	// create a client to communicate with API server.
-	clientConfig, err := config.createClientConfigFromFile()
-	if err != nil {
-		klog.Fatalf("Failed to create a ClientConfig: %v. Exiting.", err)
+	var clientConfig *restclient.Config
+	var err error
+	if config.AuthProvider == "" {
+		clientConfig, err = config.createClientConfigFromFile()
+		if err != nil {
+			klog.Fatalf("Failed to create a ClientConfig: %v. Exiting.", err)
+		}
+	} else {
+		switch config.AuthProvider {
+		case "eks":
+			if config.ClusterContextName == "" {
```

Comment:
Can we move all that stuff to a separate file?
Ideally it should bu subpackage for eks stuff.